OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -167,16 +168,24 @@ public void componentOpened() {
 
         ApplicationSettings.INSTANCE.registerSettingListeners(this);
         
-        InputStream is = null;
-        try {
-            is = getClass().getResource("/openlatexstudio/welcome.tex").openStream();
-            String welcomeMessage = IOUtils.toString(is);
-            rSyntaxTextArea.setText(welcomeMessage);
-            setDirty(true);
-        } catch (IOException ex) {
-            Exceptions.printStackTrace(ex);
-        } finally {
-            IOUtils.closeQuietly(is);
+        String initFileDir = (String) ApplicationSettings.Setting.USER_LASTFILE.getValue();
+        File initFile = new File(initFileDir);
+        if(initFile.exists() && initFile.isFile()) {
+            String content = FileService.readFromFile(initFileDir);
+            setEditorContent(content);
+            setCurrentFile(initFile);
+        } else {
+            InputStream is = null;
+            try {
+                is = getClass().getResource("/openlatexstudio/welcome.tex").openStream();
+                String welcomeMessage = IOUtils.toString(is);
+                rSyntaxTextArea.setText(welcomeMessage);
+                setDirty(true);
```

Comment:
Can we use "setEditorContent", instead of the above 2 lines for consistency inside the if?